Add upload file feature

grahn opened the pull request:
The `README.md` still needs to be updated, but otherwise this adds basic file uploading support to glack.
@ryanhatfield Let me know what you think.

grahn committed:
Change s() in main.go to sendMessage()

grahn committed:
Add initial file upload support to glack.go

grahn committed:
Add flag for uploading file to main.go

The file corresponding to the filename passed to the flag
upload-file will be uploaded.

ryanhatfield committed:
Move to version 1.0.6

Move the version from 1.0.5 to 1.0.6 for the new
upload-file feature.

ryanhatfield commented:
This works great :+1: but how would we want to handle the fact that it doesn't respect the username or icon flags? do we want to add that to the command line help, that we intentionally ignore them?

ryanhatfield committed:
Add comment on exported method

My build tools run `golint` automatically, and they
 throw a build error when an exported method is
 missing a `godoc` comment.

ryanhatfield committed:
Add json input flag and logic

ryanhatfield committed:
Merge branch 'master' into add_upload_file_feature
